src/services/image.service.ts

```diff
@@ -0,0 +1,34 @@
+import type { Playlist, PlaylistItem } from '#types/playlist';
+import { findPlaylistImageForWidth } from '#src/utils/collection';
+import type { Config } from '#types/Config';
+
+export type ImageProperty = 'shelfImage' | 'backgroundImage' | 'channelLogoImage';
+
+export const generateAlternateImageURL = (mediaid: string, imageLabel: string, width: number) =>
+  `https://img.jwplayer.com/v1/media/${mediaid}/images/${imageLabel}.jpg?width=${width}`;
+
+export const getAlternateImageLabel = (config: Config, propertyName: ImageProperty, item: PlaylistItem, playlist?: Playlist) => {
+  const alternateImageLabel = propertyName === 'shelfImage' ? playlist?.[propertyName] : item[propertyName];
+  const configImageLabel = config.custom?.[propertyName];
```

Comment:
It might be safer to decouple the string used throughout the application from the property name used in the config file in case the latter ever changes or moves, so it doesn't cause quite so much noise.  Also it obviously could cause trouble if the playlist/item property name didn't equal the custom config property name. It could also get complicated if properties move to different places outside of custom config or there's an image type that doesn't have a custom config override.

Reply:
This has been implicitly achieved since we're now mostly using the PlaylistItem properties. The actual properties are defined in an enum and only need to be changed once a different property is used.